src/EnergyPlus/Coils/CoilCoolingDXCurveFitSpeed.cc

```diff
@@ -261,8 +259,41 @@ Psychrometrics::PsychState CoilCoolingDXCurveFitSpeed::CalcSpeedOutput( Psychrom
 	outletState.h = inletState.h - ( TotCap / AirMassFlow );
 	Real64 hTinwout = inletState.h - ( ( 1.0 - SHR ) * hDelta );
 	outletState.w = Psychrometrics::PsyWFnTdbH( inletState.tdb, hTinwout );
-	outletState.tdb = Psychrometrics::PsyTdbFnHW( FullLoadOutAirEnth, FullLoadOutAirHumRat );
+	outletState.tdb = Psychrometrics::PsyTdbFnHW( outletState.h, outletState.w );
 
 	return outletState;
 
 }
+
+void CoilCoolingDXCurveFitSpeed::CalcBypassFactor(Psychrometrics::PsychState & in) {
+
+  // Bypass factors are calculated at rated conditions at sea level (make sure in.p is Standard Pressure)
+
+  // Outlet conditions
+  Psychrometrics::PsychState out;
+
+  Real64 airMassFlowRate = evap_air_flow_rate * Psychrometrics::PsyRhoAirFnPbTdbW( in.p, in.tdb, in.w);
+  Real64 deltaH = rated_total_capacity / airMassFlowRate;
+  out.p = in.p;
+  out.h = in.h - deltaH;
+  out.w = Psychrometrics::PsyWFnTdbH(out.tdb, in.h - (1.0 - gross_shr)*deltaH); // enthalpy at Tdb,in and Wout
+  out.tdb = Psychrometrics::PsyTdbFnHW(out.h, out.w);
+  out.rh = Psychrometrics::PsyRhFnTdbWPb(out.tdb, out.w, out.p);
+
+  // ADP conditions
+  Psychrometrics::PsychState adp;
+
+  adp.tdb = Psychrometrics::PsyTdpFnWPb(out.w, out.p);
+
+  Real64 tol = 1.0;
+
+  std::size_t iter = 0;
+
+  const std::size_t maxIter(50);
+
+  while( (iter <= maxIter) ) {
+
```

Comment:
@Myoldmopar This is where I was when I was re-writing the bypass factor calculation. You can reference the CalcBypassFactor method in either DXCoils or VariableSpeedCoils (though they are slightly different so you'll want to take the best from both).